core/arch/instrlist.c

```diff
@@ -245,7 +245,8 @@ check_translation(instrlist_t *ilist, instr_t *inst)
 #if defined(CLIENT_INTERFACE) && defined(ARM)
     if (instr_is_meta(inst)) {
         dr_pred_type_t auto_pred = ilist->auto_pred;
-        if (auto_pred != DR_PRED_AL && auto_pred != DR_PRED_NONE) {
+        if (auto_pred != DR_PRED_AL && auto_pred != DR_PRED_OP &&
```

Comment:
BTW since you'll be adding more commits here I hope you don't mind if I cancel your CI builds for 865ef9e to shrink the queue for my imminent PR...